snakemake/deployment/conda.py

```
@@ -250,16 +260,17 @@ def address(self):
 
         """
-        if self.is_named:
+        if self.name:
             return self.name
+        elif self.dir:
+            return self.dir
         else:
-            hash = self.hash
-            env_dir = self._env_dir
+            env_dir = self._envs_dir
             get_path = lambda h: os.path.join(env_dir, h)
```

Ruff

Do not assign a `lambda` expression, use a `def`

Rewrite `get_path` as a `def`

(E731)